### PR TITLE
[Standalone Agent] Disallow upgrade if upgrade is already in progress

### PR DESCRIPTION
**Pull request by ycombinator**
## What does this PR do?

This PR prevents users of standalone Agents from being able to upgrade the Agent if an upgrade is already in progress.

## Why is it important?

Attempting to upgrade an Agent while an upgrade is already in progress could corrupt the Agent installation.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~I have made corresponding changes to the documentation~
- [ ] ~I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [x] I have added an integration test or an E2E test

## Related issues

- Relates #2706

### CODE REVIEW

File: pkg/control/v2/client/mocks/client.go
Comment (pchila) [inline pkg/control/v2/client/mocks/client.go:1]: :heart:

File: pkg/control/v2/client/client.go [hunk 1]
```diff
@@ -156,6 +156,7 @@ type DiagnosticComponentResult struct {
 }
 
 // Client communicates to Elastic Agent through the control protocol.
+// go:generate mockery --name Client
```
Comment (pchila) [inline pkg/control/v2/client/client.go:159]: :+1:

**Review by pchila — COMMENTED** [review]
Code looks good. 
There is a clash between your changes for the upgrade integration test and the modifications proposed in https://github.com/elastic/elastic-agent/pull/3477 so I guess we have to sync a bit to make sure we don't break/lose testcases

For the code coverage, I guess we have to exclude `mocks/*.go` from analysis since most of the uncovered lines are actually the generated code.

**Comment by ycombinator** [comment]
> For the code coverage, I guess we have to exclude `mocks/*.go` from analysis since most of the uncovered lines are actually the generated code.

Done in 98764d81a735503681812e44487ad1043a5cb4c0.

**Review by pchila — APPROVED** [review]

**Pull request merged**